### ISSUE TITLE
[REVIEW]: Kinetic.jl: A lightweight finite volume toolbox in Julia

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3060]
**Submitting author:** @vavrines (<a href="http://orcid.org/0000-0001-9127-9497">Tianbai Xiao</a>)
**Repository:** <a href="https://github.com/vavrines/Kinetic.jl" target ="_blank">https://github.com/vavrines/Kinetic.jl</a>
**Version:** v0.7.5
**Editor:** @diehlpk
**Reviewers:** @rdeits, @jarvist
**Archive:** <a href="https://doi.org/10.5281/zenodo.4958132" target="_blank">10.5281/zenodo.4958132</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/65d56efef938caf92c2cc942d2c25ea4/status.svg)](https://joss.theoj.org/papers/65d56efef938caf92c2cc942d2c25ea4)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/65d56efef938caf92c2cc942d2c25ea4"><img src="https://joss.theoj.org/papers/65d56efef938caf92c2cc942d2c25ea4/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/65d56efef938caf92c2cc942d2c25ea4/status.svg)](https://joss.theoj.org/papers/65d56efef938caf92c2cc942d2c25ea4)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/vavrines/Kinetic.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

 @jarvist & @rdeits, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @diehlpk know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/vavrines/Kinetic.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@vavrines) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/vavrines/Kinetic.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@vavrines) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @KristofferC, @rdeits it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.05 s (1223.4 files/s, 97969.2 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Markdown                        42            257              0           1862
TOML                             4            260              1           1218
Julia                           13            170            237            972
YAML                             5              7             30            115
TeX                              1              0              0             76
-------------------------------------------------------------------------------
SUM:                            65            694            268           4243
-------------------------------------------------------------------------------


Statistical information for the repository 'e631d825b06217276c947127' was
gathered on 2021/02/23.
No commited files with the specified extensions were found.
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.2307/3609795 may be a valid DOI for title: The mathematical theory of non-uniform gases: an account of the kinetic theory of viscosity, thermal conduction and diffusion in gases
- 10.1137/141000671 may be a valid DOI for title: Julia: A fresh approach to numerical computing
- 10.1016/j.jcp.2016.12.022 may be a valid DOI for title: A well-balanced unified gas-kinetic scheme for multiscale flow transport under gravitational field

INVALID DOIs

- https://doi.org/10.5281/zenodo.4025432 is INVALID because of 'https://doi.org/' prefix
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by diehlpk** [comment]
@whedon remind @rdeits  in 2 weeks

**Comment by whedon** [comment]
@rdeits  doesn't seem to be a reviewer or author for this submission.

**Comment by diehlpk** [comment]
@whedon remind @KristofferC in 2 weeks

**Comment by whedon** [comment]
Reminder set for @KristofferC in 2 weeks

**Comment by whedon** [comment]
:wave: @KristofferC, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @rdeits, please update us on how your review is going (this is an automated reminder).

**Comment by diehlpk** [comment]
Reminder set for @KristofferC in 2 weeks

**Comment by diehlpk** [comment]
@whedon remind @rdeits in 2 weeks

**Comment by whedon** [comment]
Reminder set for @rdeits in 2 weeks

**Comment by whedon** [comment]
:wave: @KristofferC, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @rdeits, please update us on how your review is going (this is an automated reminder).

**Comment by rdeits** [comment]
I've submitted an issue to Kinetic.jl with some notes on the installation + setup process: https://github.com/vavrines/Kinetic.jl/issues/12

**Comment by diehlpk** [comment]
@whedon remind @KristofferC in 2 weeks

**Comment by diehlpk** [comment]
@whedon remind @rdeits in 2 weeks

**Comment by whedon** [comment]
Reminder set for @rdeits in 2 weeks

**Comment by rdeits** [comment]
Thanks for the reminder. I'm still having trouble getting the basic example to run at all--I've updated the discussion here: https://github.com/vavrines/Kinetic.jl/issues/12#issuecomment-822746390

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by vavrines** [comment]
@rdeits Sorry for the inconvenience. I've revised the sample code in the manuscript, please check the latest version.

**Comment by vavrines** [comment]
@diehlpk 
Dear Patrick, May I have a question here?
I found the JOSS doesn't recognize my markdown bullet points at the end of first page, i.e.
```
- 100% Julia stack that encounters no two-language problem
- Comprehensive support for kinetic theory and phase-space equations
- ...
```
Does JOSS support any format of bullet points, or I have to reformat it into plain text?
Best,

**Comment by diehlpk** [comment]
> @diehlpk
> Dear Patrick, May I have a question here?
> I found the JOSS doesn't recognize my markdown bullet points at the end of first page, i.e.
> 
> ```
> - 100% Julia stack that encounters no two-language problem
> - Comprehensive support for kinetic theory and phase-space equations
> - ...
> ```
> 
> Does JOSS support any format of bullet points, or I have to reformat it into plain text?
> Best,

@vavrines Could you please try the star (*) instead of the dash (-). I think that the star should work.

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by vavrines** [comment]
> > @diehlpk
> > Dear Patrick, May I have a question here?
> > I found the JOSS doesn't recognize my markdown bullet points at the end of first page, i.e.
> > ```
> > - 100% Julia stack that encounters no two-language problem
> > - Comprehensive support for kinetic theory and phase-space equations
> > - ...
> > ```
> > 
> > 
> > Does JOSS support any format of bullet points, or I have to reformat it into plain text?
> > Best,
> 
> @vavrines Could you please try the star (*) instead of the dash (-). I think that the star should work.

It seems star doesn't work either.

**Comment by diehlpk** [comment]
@vavrines We used in our JOSS paper both the dash and star. Please have a look at the Application section 

https://github.com/STEllAR-GROUP/hpx/blob/master/docs/joss_paper/paper.md

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by vavrines** [comment]
> @vavrines We used in our JOSS paper both the dash and star. Please have a look at the Application section
> 
> https://github.com/STEllAR-GROUP/hpx/blob/master/docs/joss_paper/paper.md
> 
> Can you try to have a blank line and indent the dash or star?

With blank line it works, thx a lot!

**Comment by diehlpk** [comment]
Hey @KristofferC how is your review going?

**Comment by diehlpk** [comment]
@whedon remind @rdeits in 2 weeks

**Comment by whedon** [comment]
Reminder set for @rdeits in 2 weeks

**Comment by whedon** [comment]
:wave: @KristofferC, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @rdeits, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
OK, @KristofferC is no longer a reviewer

**Comment by diehlpk** [comment]
Hey @rdeits when do you estimate to finish the review?

**Comment by rdeits** [comment]
I was able to get all the examples running properly, so I should be able to
finish this weekend.

**Comment by rdeits** [comment]
Could one of the `openjournals` admins confirm that I still have collaborator access on this repository? I can no longer modify the checklist above, and I do not see this repo listed under my https://github.com/settings/repositories list anymore. 

I know that I previously had access because I was able to partially complete the checklist earlier.

**Comment by vavrines** [comment]
Hi @diehlpk
Could you please help one reviewer on the checklist and maybe remind another reviewer? (I'm not sure whether he is aware of the existence of this review issue) : )

**Comment by diehlpk** [comment]
@whedon add @rdeits as reviewer

**Comment by whedon** [comment]
OK, @rdeits is now a reviewer

**Comment by diehlpk** [comment]
@rdeits Please try now

**Comment by diehlpk** [comment]
> Hi @diehlpk
> Could you please help one reviewer on the checklist and maybe remind another reviewer? (I'm not sure whether he is aware of the existence of this review issue) : )

@vavrines Sure

**Comment by diehlpk** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon recommend-accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by diehlpk** [comment]
@whedon re-invite @rdeits  as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@rdeits  please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by diehlpk** [comment]
@rdeits Please click on the link above and you should be able to edit the checkboxes again.

**Comment by diehlpk** [comment]
@rdeits Is it working for you now? Note that the above links expires in two weeks again.

**Comment by diehlpk** [comment]
Hey  @jarvist how is your review going?

**Comment by jarvist** [comment]
@diehlpk - I think I failed to catch the invite to edit in time. Could you re-invite me?
I've installed the software and checked functionality, I haven't read the documentation or paper yet. I should have this done by early next week.

**Comment by diehlpk** [comment]
@whedon re-invite @jarvist as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@jarvist please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by diehlpk** [comment]
@jarvist Please click on the link above.

**Comment by jarvist** [comment]
@diehlpk  - thank you! All sorted now. I'll have this done next week.

**Comment by rdeits** [comment]
@diehlpk yes, thanks--the invite worked and I'm now able to edit the checklist again. I'm not sure how I lost access to begin with, but I'm glad it's all sorted now.

**Comment by rdeits** [comment]
Ok, I think this is in good shape. I made a couple of final requests here: https://github.com/vavrines/Kinetic.jl/issues/12#issuecomment-855258289 (mostly just minor typos for grammatical issues), and I've checked off everything else in the list.

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3060 with the following error: 
```
 [WARNING] Citeproc: citation jarvist not found
Error producing PDF.
! Undefined control sequence.
\hyper@@link ->\let \Hy@reserved@a 
                                   \relax \@ifnextchar [{\hyper@link@ }{\hyp...
l.385 }

Looks like we failed to compile the PDF

```

**Comment by vavrines** [comment]
@whedon check repository

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.11 s (601.3 files/s, 45726.2 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Markdown                        43            278              0           1957
TOML                             5            265              9           1254
Julia                           11            125            131            709
SVG                              2              0              2            126
YAML                             5              7             30            117
TeX                              1              0              0             85
-------------------------------------------------------------------------------
SUM:                            67            675            172           4248
-------------------------------------------------------------------------------


Statistical information for the repository '6ef1a2b07759fc0f2c2c9c00' was
gathered on 2021/06/07.
No commited files with the specified extensions were found.
```

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3060 with the following error: 
```
 [WARNING] Citeproc: citation jarvist not found
Error producing PDF.
! Undefined control sequence.
\hyper@@link ->\let \Hy@reserved@a 
                                   \relax \@ifnextchar [{\hyper@link@ }{\hyp...
l.385 }

Looks like we failed to compile the PDF

```

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3060 with the following error: 
```
 [WARNING] Citeproc: citation jarvist not found
Error producing PDF.
! Undefined control sequence.
\hyper@@link ->\let \Hy@reserved@a 
                                   \relax \@ifnextchar [{\hyper@link@ }{\hyp...
l.385 }

Looks like we failed to compile the PDF

```

**Comment by vavrines** [comment]
Hi @diehlpk

I have a problem compiling my paper, which is related to the LaTex package `{hyperref}`.
I didn't cite a `jarvist` in my markdown file nor in the bibtex.
It seems that whedon somehow misunderstood the reviewer as a citation.
Do you have an idea on how to fix it?

**Comment by diehlpk** [comment]
@openjournals/dev Could on please have a look at the compilation error?

**Comment by diehlpk** [comment]
> Hi @diehlpk
> 
> I have a problem compiling my paper, which is related to the LaTex package `{hyperref}`.
> I didn't cite a `jarvist` in my markdown file nor in the bibtex.
> It seems that whedon somehow misunderstood the reviewer as a citation.
> Do you have an idea on how to fix it?

@vavrines I asked the support team for help, since I do not know what is going wrong.

**Comment by arfon** [comment]
> Submitting author: @vavrines (Tianbai Xiao)
> Repository: https://github.com/vavrines/Kinetic.jl
> Version: v0.7.0
> Editor: @diehlpk
> Reviewers: @rdeits @jarvist, @rdeits

The issue here is that the `Reviewers:` string `@rdeits @jarvist, @rdeits` is missing a comma, and so Whedon is passing `@jarvist` as a variable into the compilation step which is being interpreted as a citation.

So, the fix here is to add a comma between `@rdeits @jarvist` in the reviewer list above (which I'm going to do now).

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by vavrines** [comment]
@whedon set v0.7.4 as version

**Comment by whedon** [comment]
I'm sorry @vavrines, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by rdeits** [comment]
I've completed my review checklist, and the submission looks good to me.

**Comment by diehlpk** [comment]
@rdeits Thanks, I just checked, and it seems that `Quality of writing` is unchecked. Could you please have a look?

**Comment by diehlpk** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon recommend-accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by diehlpk** [comment]
@whedon remind @jarvist in 2 weeks

**Comment by whedon** [comment]
Reminder set for @jarvist in 2 weeks

**Comment by rdeits** [comment]
Ah, sorry. Fixed now

**Comment by jarvist** [comment]
I seem to have lost my 'collaborator' badge, and so cannot tick off things on my checklist. 
I'm just reviewing this package and paper now, so will make a note separately as I go.

**Comment by jarvist** [comment]
@vavrines - it's a bit suboptimal that Kinetic (rather KitBase) has a dependency on Plots. Currently this has actually broken precompilation on my machine, due to some kind of interaction with the way Plots is being initialised & attempting to pull in GR (which is a binary package, and breaks not irregularly). 

It would require quite a lot of refactoring, but using PlotsRecipes avoids this issue, and makes the dependency lighter to install.
https://docs.juliaplots.org/latest/recipes/

**Comment by jarvist** [comment]
@vavrines Could you please confirm that the example given in the JOSS paper works OK? 
I'm currently getting:
```
; julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.1 (2021-04-23)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Kinetic

julia> set, ctr, xface, yface, t = Kinetic.initialize("kinetic.toml")
==============================================================
Kinetic.jl
A Lightweight Toolbox for Kinetic Modeling and Simulation
==============================================================

reading configurations from kinetic.toml

initializing solver: 
ERROR: MethodError: no method matching iterate(::Nothing)
Closest candidates are:
  iterate(::Union{LinRange, StepRangeLen}) at range.jl:664
  iterate(::Union{LinRange, StepRangeLen}, ::Int64) at range.jl:664
  iterate(::T) where T<:Union{Base.KeySet{var"#s79", var"#s78"} where {var"#s79", var"#s78"<:Dict}, Base.ValueIterator{var"#s77"} where var"#s77"<:Dict} at dict.jl:693
  ...
Stacktrace:
 [1] indexed_iterate(I::Nothing, i::Int64)
   @ Base ./tuple.jl:89
 [2] top-level scope
   @ REPL[2]:1

julia> 
```

**Comment by vavrines** [comment]
> @vavrines Could you please confirm that the example given in the JOSS paper works OK?
> I'm currently getting:
> 
> ```
> ; julia
>                _
>    _       _ _(_)_     |  Documentation: https://docs.julialang.org
>   (_)     | (_) (_)    |
>    _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
>   | | | | | | |/ _` |  |
>   | | |_| | | | (_| |  |  Version 1.6.1 (2021-04-23)
>  _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
> |__/                   |
> 
> julia> using Kinetic
> 
> julia> set, ctr, xface, yface, t = Kinetic.initialize("kinetic.toml")
> ==============================================================
> Kinetic.jl
> A Lightweight Toolbox for Kinetic Modeling and Simulation
> ==============================================================
> 
> reading configurations from kinetic.toml
> 
> initializing solver: 
> ERROR: MethodError: no method matching iterate(::Nothing)
> Closest candidates are:
>   iterate(::Union{LinRange, StepRangeLen}) at range.jl:664
>   iterate(::Union{LinRange, StepRangeLen}, ::Int64) at range.jl:664
>   iterate(::T) where T<:Union{Base.KeySet{var"#s79", var"#s78"} where {var"#s79", var"#s78"<:Dict}, Base.ValueIterator{var"#s77"} where var"#s77"<:Dict} at dict.jl:693
>   ...
> Stacktrace:
>  [1] indexed_iterate(I::Nothing, i::Int64)
>    @ Base ./tuple.jl:89
>  [2] top-level scope
>    @ REPL[2]:1
> 
> julia> 
> ```

Hi @jarvist 

It shoud be fine. The CI test passed here (https://github.com/vavrines/Kinetic.jl/blob/master/test/runtests.jl).
Could you please check the version of Kinetic?
There was a bug in v0.7.0 and has been fixed in the latest version.

**Comment by vavrines** [comment]
@jarvist 
I Just tested the code again. Actually I never saw such error beforehand. It seems that Julia doesn't recognize any text in the config file (if it's not empty).
I'm not sure how you generate the config (maybe an encoding thing?), but I guess a copy-paste of the CI config file will help get rid of this.
Please let me know if it doesn't work.

**Comment by jarvist** [comment]
@vavrines Thanks! It seems I ended up pinned on an old version of Kinetic due to version incompatibility with other installed packages. I've got that sorted now, and the example from the JOSS (copied + pasted out of the PDF, as I assumed that someone would try that at some point...) seems to be solving.

**Comment by danielskatz** [comment]
@whedon re-invite @jarvist as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@jarvist please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by jarvist** [comment]
OK! All looking good. 
Two minor issues:-
 * the 'community guidelines' are OK, but a little unclear. How should people get in touch? via a Github issue? Personal email? etc. 
 * the aforementioned dependency on Plots.jl; this would be better refactored to use Plots Recipes 

But I don't think either of these things should block publication.

**Comment by vavrines** [comment]
> OK! All looking good.
> Two minor issues:-
> 
> * the 'community guidelines' are OK, but a little unclear. How should people get in touch? via a Github issue? Personal email? etc.
> * the aforementioned dependency on Plots.jl; this would be better refactored to use Plots Recipes
> 
> But I don't think either of these things should block publication.

Thanks for the input! I'm gonna make the point clearer.
The RecipesBase idea sounds interesting. I'm not sure whether it's the best fit here, since the `plot_line` and `plot_contour` functions need to use a plot function anyway. I'll look into it.

**Comment by diehlpk** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.2307/3609795 may be a valid DOI for title: The mathematical theory of non-uniform gases: an account of the kinetic theory of viscosity, thermal conduction and diffusion in gases
- 10.1137/141000671 may be a valid DOI for title: Julia: A fresh approach to numerical computing
- 10.1016/j.jcp.2016.12.022 may be a valid DOI for title: A well-balanced unified gas-kinetic scheme for multiscale flow transport under gravitational field
- 10.1016/j.jcp.2020.109535 may be a valid DOI for title: A velocity-space adaptive unified gas kinetic scheme for continuum and rarefied flows

INVALID DOIs

- https://doi.org/10.5281/zenodo.4025432 is INVALID because of 'https://doi.org/' prefix
```

**Comment by diehlpk** [comment]
@vavrines Please fix the invalid DOI reported above.

**Comment by diehlpk** [comment]
@whedon generate paper

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by diehlpk** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by diehlpk** [comment]
@rdeits and @jarvist thanks for reviewing for JOSS!

**Comment by diehlpk** [comment]
@vavrines Please address my editorial issues below and we are good to prepare the final paper. 

1. Karlsruhe Institute of Technology -> Please add the city and country to the affiliation
2.  `For example, it is cumbersome to implement the phase-23field evolution from the Boltzmann equation (Xiao et al., 2017,2020) in OpenFOAM or24integrate it with scientific machine learning (SciML) package` Please elaborate here and give some explanation why it is cumbersome.
3.  `Besides, the two-language problem introduces additional trade-off in both development and execution` Please elaborate on the trade off.
4. I think some reference to Julia itself would be beneficial for the reader. maybe> https://github.com/JuliaLang/julia/blob/master/CITATION.bib

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3060 with the following error: 
```
 Error reading bibliography file paper.bib:
(line 6, column 3):
unexpected "d"
expecting space, ",", white space or "}"
Looks like we failed to compile the PDF

```

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by vavrines** [comment]
@whedon check references

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2307/3609795 is OK
- 10.1017/CBO9780511800955 is OK
- 10.1016/j.cpc.2016.11.010 is OK
- 10.1137/141000671 is OK
- 10.5281/zenodo.4025432 is OK
- 10.1016/j.jcp.2016.12.022 is OK
- 10.1016/j.jcp.2020.109535 is OK
- 10.1016/j.camwa.2020.04.033 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by vavrines** [comment]
Hi @diehlpk 

Thank you very much for the suggestions.
I revised the paper with the incorporation of the editorial issues.
The major changes include:

- The city and country were added to the affiliation.
- The two points were reorganized and elaborated to avoid misunderstanding.
- A Julia paper was already cited in the first version, and I added the original paper in 2012 in the bibliography.

Please have a look at the revision and let me know if there are further ambiguities.

**Comment by diehlpk** [comment]
@vavrines Thanks, that looks good.

**Comment by diehlpk** [comment]
@vavrines I found some typos in the latest version of the paper

* cubersome -> cumbersome
* optmizing - > optimizing
* vectorizaion -> vectorization 

Please fix them before we can go forward.

**Comment by vavrines** [comment]
@diehlpk Thanks! fixed

**Comment by vavrines** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by diehlpk** [comment]
@vavrines 

- [ ] Generate a new release with all fixed in the paper and the code made during the review. Please post the release number here.
- [ ] Upload the release to Zendo or Figshare and post the DOI here.

**Comment by vavrines** [comment]
@diehlpk 

- The new release number is v0.7.5 (https://github.com/vavrines/Kinetic.jl/releases/tag/v0.7.5), with the title same as paper.
- The Zenodo record is here (https://zenodo.org/record/4958132#.YMjH9C0RppR)

**Comment by diehlpk** [comment]
@when commands

**Comment by diehlpk** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon recommend-accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by diehlpk** [comment]
@whedon set v0.7.5 as version

**Comment by whedon** [comment]
OK. v0.7.5 is the version.

**Comment by diehlpk** [comment]
@vavrines 

`vavrines/Kinetic.jl: Kinetic.jl: A portable finite volume toolbox for scientific and neural computing`

The Zendo title does not match the title of the paper. Please update the title there as well. I recommend to add your DOI asl well.

**Comment by diehlpk** [comment]
@whedon set  10.5281/zenodo.4958132 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4958132" target="_blank">10.5281/zenodo.4958132</a> is the archive.

**Comment by diehlpk** [comment]
@vavrines Perfect, I can not recommend your paper for acceptance.

**Comment by diehlpk** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2307/3609795 is OK
- 10.1017/CBO9780511800955 is OK
- 10.1016/j.cpc.2016.11.010 is OK
- 10.1137/141000671 is OK
- 10.5281/zenodo.4025432 is OK
- 10.1016/j.jcp.2016.12.022 is OK
- 10.1016/j.jcp.2020.109535 is OK
- 10.1016/j.camwa.2020.04.033 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2391

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2391, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@vavrines - While proofreading this paper before publishing it, I found a number of things that I suggested fixes for in https://github.com/vavrines/Kinetic.jl/pull/15  Please merge this or let me know what you disagree with, then we can proceed with the final publication.

**Comment by vavrines** [comment]
> @vavrines - While proofreading this paper before publishing it, I found a number of things that I suggested fixes for in [vavrines/Kinetic.jl#15](https://github.com/vavrines/Kinetic.jl/pull/15) Please merge this or let me know what you disagree with, then we can proceed with the final publication.

Hi @danielskatz 

Thank you very much for the careful check and revision.
The patch has been merged into master branch.
Do I need to make another release on Zenodo for the latest change?

**Comment by danielskatz** [comment]
no, you don't need to make a new release for changes in the paper/bib

**Comment by danielskatz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03060/joss.03060/10.21105.joss.03060.pdf) :page_facing_up: :point_left:

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1404861823275483142) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2392
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03060](https://doi.org/10.21105/joss.03060)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @vavrines (Tianbai Xiao)!!

And thanks to @rdeits and @jarvist for reviewing and @diehlpk editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03060/status.svg)](https://doi.org/10.21105/joss.03060)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03060">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03060/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03060/status.svg
   :target: https://doi.org/10.21105/joss.03060
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03060/status.svg)](https://doi.org/10.21105/joss.03060)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
